website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,237 @@
+                {
+                    'curr_product_id': curr_product,
+                    'curr_types': curr_types,
+                    'curr_chd_price_component_ats':
+                    curr_chd_price_component_ats,
+                    'avail_accessories': avail_accessories,
+                    })
+        # first iteration, loading the page
+        return request.render('website_chd_product_configurator.conf_start', {
+            'conf_products': Conf_products.search([(
+                'chd_origin_product', '=', True)
+                ]),
+            })
+
+    # shows the options list
+    @http.route('/chd_init/<id>/', website=True)
+    def call_configurator(self, **form_data):
+        errormsg = ""
+        Conf_products = request.env['product.template']
+        chd_results = request.env['chd.product_configurator.result']
+        curr_product_id = Conf_products.search([('id', '=', form_data['id'])])
+        chd_price_components_at = request.env[
+            'price.component.attribute.template']
+        all_accessories = []
+        all_attributes = {}
+        chd_dict = {
+            'origin_product_id': curr_product_id.ids[0],
+            'partner_id': self.get_current_partner().id,
+            'state': 'config',
+            'quantity': form_data['quantity']
+            }
+        # dynamic and fixed size types
+        if form_data['product_id_chd_size_type'] == "fixed":
+            chd_dict['size_id'] = form_data['size']
+            chd_size = request.env['chd.size'].search([
+                ('id', '=', form_data['size'])
+                ])
+            chd_dict['width'] = chd_size.width
+            chd_dict['height'] = chd_size.height
+        else:
+            chd_dict['width'] = form_data['width']
+            chd_dict['height'] = form_data['height']
+        # if there aren't any errors, upload the image
+        message = ''
+        has_image = Conf_products.search([
+            ('id', '=', form_data['product_id'])
+            ])[0].chd_configurator_has_image
+        if has_image:
+            try:
+                import base64
+                fileitem = form_data['pic']
+                # add uploaded image to configurator
+                chd_dict['image'] = base64.b64encode(fileitem.stream.read())
+                chd_dict['image_filename'] = fileitem.filename
+                message = 'image uploaded successfully'
+            except:
+                message = 'Problems uploading your image, please contact us'
+        if message == '':
+            message = 'no image needed for this product'
+        for key in form_data:
+            if ('pricecomponent_' in key):
+                # if it is of type "string" the value is the index of the
+                # selection in pricecomponent_name, the id encoded in the key
+                if 'pricecomponent_string' in key:
+                    pricecomponent_id = int(key.split('_')[3])
+                    pricecomponent_value = form_data[key]
```

Comment:
price components are of two types "string" and " numeric", if it is "string" it is renedered as a dropdown select, if it is "numeric" it is rendered in the front end as int input. just checked , it  is a leftover peice of code from when the two cases  where encoded differently , definetely not needed anymore.